grant-project/grant-monorepo
# Profile edit

Closes #145.

### What this does
- Profile edit UI
- basic backend `/users/<id>` PUT update functionality with tests

### What this doesn't do
- backend user auth check
- custom backend field validation 
- image upload
- borrows heavily from create flow TeamMember edit, but differs enough for me to hesitate DRYing them together, I suspect they will diverge further in the future

### Test
- visit any user other than authenticated user and ensure "Edit profile" doesn't appear
- visit authenticated user profile, click "Edit profile" button
- observe that the "Save changes" button only becomes enabled when something has been changed
- check that all editable fields update on save
- try adding image and deleting image

### Screencaps
<img width="903" alt="screen shot 2018-10-22 at 10 55 34 pm" src="https://user-images.githubusercontent.com/11430954/47335806-90cfcd00-d652-11e8-8489-01e427432cc4.png">
<img width="897" alt="screen shot 2018-10-22 at 10 55 53 pm" src="https://user-images.githubusercontent.com/11430954/47335810-95948100-d652-11e8-8256-6dcbaf5418de.png">
<img width="329" alt="screen shot 2018-10-22 at 10 56 53 pm" src="https://user-images.githubusercontent.com/11430954/47335817-9a593500-d652-11e8-9078-9fb2fa050630.png">
